pkg/api/pod/util.go

```diff
@@ -260,3 +263,11 @@ func DropDisabledVolumeMountsAlphaFields(volumeMounts []api.VolumeMount) {
 		}
 	}
 }
+
+// DropDisabledVolumeDevicesAlphaFields removes disabled fields from []VolumeDevice.
+// This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a VolumeDevice
+func DropDisabledVolumeDevicesAlphaFields(volumeDevices []api.VolumeDevice) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolumeSupport) {
+		volumeDevices = nil
```

Comment:
I don't think this works.  You'll need to pass in the Container object and set VolumeDevices to nil.  Or pass in the ptr.

Reply:
Should we add test cases for these drop methods?

Reply:
done...added a test case for the VolumeDevices in pod, and VolumeMode for pv and pvc - unit tests seem to be validating fine